test/unit/file-utilities.js

```diff
@@ -0,0 +1,150 @@
+    , 'subdir': {
+        'somelog2.txt': logEntry
+      }
+    })
+
+    fileUtilities.getFiles({}, path.join(tempDir, '*.txt'), (err, files) => {
+      tt.error(err, 'No error')
+      tt.type(files, Array, 'files is an array')
+      tt.equal(files.length, 1, 'File count is correct')
+      tt.deepEqual(files[0], path.join(tempDir, 'somelog1.txt'), 'Filename is correct')
+      tt.end()
+    })
+  })
+
+  t.test('retrieves 2 files based on glob pattern **/*.txt', (tt) => {
+    const tempDir = tt.testdir({
+      'somelog1.txt': logEntry
+    , 'subdir': {
+        'somelog2.txt': logEntry
+      , 'somelog3.log': logEntry // Should not match due extension mismatch
+      , 'somelog4': logEntry // Should not match because extensionless is not defined
+      }
+    })
+
+    fileUtilities.getFiles({}, path.join(tempDir, '**', '*.txt'), (err, files) => {
+      tt.error(err, 'No error')
+      tt.type(files, Array, 'files is an array')
+      tt.equal(files.length, 2, 'File count is correct')
+      tt.deepEqual(files.sort(), [
+        path.join(tempDir, 'somelog1.txt')
+      , path.join(tempDir, 'subdir', 'somelog2.txt')
+      ].sort(), 'Filenames are correct')
+      tt.end()
+    })
+  })
+
+  t.test('if a file is given, it returns a single file list', (tt) => {
+    const tempDir = tt.testdir({
+      'singlefile.whatever': logEntry
+    })
+
+    const file = path.join(tempDir, 'singlefile.whatever')
+    fileUtilities.getFiles({}, file, (err, files) => {
+      tt.error(err, 'No error')
+      tt.type(files, Array, 'files is an array')
+      tt.equal(files.length, 1, 'File count is correct')
+      tt.deepEqual(files, [file], 'Single filename is correct')
+      tt.end()
+    })
+  })
+
+  t.test('Error: Directory doesn\'t exist', (tt) => {
+    fileUtilities.getFiles({}, 'DOESNOTEXIST', (err, files) => {
+      tt.type(err, Error, 'Expected error received')
+      tt.match(err, {
+        name: 'Error'
+      , message: /no such file or directory/
+      , code: 'ENOENT'
+      , path: 'DOESNOTEXIST'
+      })
+      tt.end()
+    })
+  })
+
+  t.test('Error: special files (sockets, block devs) return an error', (tt) => {
+    fileUtilities.getFiles({}, '/dev/null', (err, files) => {
+      tt.type(err, Error, 'Expected error received')
+      tt.match(err, {
+        name: 'Error'
+      , message: 'not a file or directory'
+      })
+      tt.end()
+    })
+  })
+
+  t.test('Error: `glob` returns an error', (tt) => {
+    const readdir = fs.readdir
+    tt.on('end', async () => {
+      fs.readdir = readdir
+    })
+    fs.readdir = function(path, cb) {
+      process.nextTick(function() {
+        cb(new Error('mock fs.readdir error'))
+      })
+    }
+    fileUtilities.getFiles({}, '/tmp/*.something', (err, files) => {
+      tt.type(err, Error, 'Expected error received')
+      tt.match(err, {
+        name: 'Error'
+      , message: 'mock fs.readdir error'
+      })
+      tt.end()
+    })
+  })
+})
```

Comment:
Might be worth adding tests for symlinks and hardlinks?

Reply:
Sure, I can put a symlink in there

Reply:
done!